sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/RetryUtilTest.java

```diff
@@ -108,7 +108,7 @@ void withRetryFluxEmitsItemsLaterThanTimeout() {
             .doOnSubscribe(s -> resubscribe.incrementAndGet());
 
         // Act & Assert
-        StepVerifier.create(RetryUtil.withRetry(flux, options, timeoutMessage))
+        StepVerifier.withVirtualTime(() -> RetryUtil.withRetry(flux, options, timeoutMessage))
```

Comment:
Watch out around `withVirtualTime` as it doesn't always behave the best in a multi-threaded environment. From what I've seen in the past:

> StepVerifier.withVirtualTime is a great way to mock execution through time, for example mocking 30 minutes of "running" with a simple API that takes milliseconds to complete. Unfortunately, the base overload uses a shared Scheduler for running which can result in states where the scheduler is shutdown or isn't instantiated when the test runs. So, if you need to use this API you can do one of two things:
> 	
> Pass an instance Scheduler instead of using the shared Scheduler.
>	
> Annotated the test class with @Isolated and @Execution(ExecutionMode.SAME_THREAD).  (more notes on these later)